### PR TITLE
Refactor WiX installer projects

### PR DESCRIPTION
**Pull request by barnson**

### CODE REVIEW

File: platforms/Windows/bld/bld.wixproj
Comment (compnerd) [inline platforms/Windows/bld/bld.wixproj:7]: Not sure if there is a better way to do this, IIRC this was required for the use of heat.

File: platforms/Windows/bld/bld.wxs
Comment (compnerd) [inline platforms/Windows/bld/bld.wxs:28]: Would be great to know if you have an answer to this question - is there a way to use a symbolic link that can be laid down rather than duplicating the binary itself.

File: platforms/Windows/bld/bld.wxs
Comment (compnerd) [inline platforms/Windows/bld/bld.wxs:331]: Would be good to know if we should preserve this - this is meant to support the dual-scope installers.

File: platforms/Windows/bundle/installer.wixproj
Comment (compnerd) [inline platforms/Windows/bundle/installer.wixproj:4]: Curious about the `-X-` infix here, what is the reasoning for that?

File: platforms/Windows/bundle/installer.wxs
Comment (compnerd) [inline platforms/Windows/bundle/installer.wxs:19]: The other locations indent the continuation, would be good to maintain the style.

```suggestion
        LicenseUrl="https://www.swift.org/LICENSE.txt"
        LogoFile="swift.png"
        SuppressOptionsUI="yes"
        SuppressRepair="no"
        Theme="hyperlinkLicense" />
```

File: platforms/Windows/bld/bld.wxs
Comment (compnerd) [inline platforms/Windows/bld/bld.wxs:10]: What does the `{0}` signify/do?

File: platforms/Windows/shared/swift.en-us.wxl
Comment (compnerd) [inline platforms/Windows/shared/swift.en-us.wxl:16]: Nit: missing newline

File: platforms/Windows/bundle/installer.wxs
Comment (compnerd) [inline platforms/Windows/bundle/installer.wxs:29]: Unfortunately, these are not yet generated in all builds.  If they were, I think that we would want to put `swift-format` into `cli.msi` and `swift-inspect` into `dbg.msi`.  Perhaps we can just merge the entries there with an option to just warn if the file is not present and continuing?

Comment (barnson) [reply platforms/Windows/bundle/installer.wxs:29]: We can conditionalize on the architecture. Do you want me to move them now? It's just as easy to conditionalize in the bundle as in a package?

File: platforms/Windows/bundle/installer.wxs
Comment (tristanlabelle) [inline platforms/Windows/bundle/installer.wxs:7]: Is it no problem to have the same UpgradeCode for both architectures?

Would that prevent installing both the arm64 and amd64 binaries on an arm64 machine (which supports amd64 emulation)? (maybe not a real scenario, asking to understand)

File: platforms/Windows/bundle/installer.wxs
Comment (compnerd) [reply platforms/Windows/bundle/installer.wxs:30]: I think that we can just move them now as long as they don't break the build.  They are intended to be built always and so rather than increasing the number of MSIs further we can reduce them a bit and possibly reduce the amount of code for this.  The preprocessor and property should avoid the MSI for the single entry (the project references make the MSIs seem even sillier than when they are standalone).

Comment (barnson) [reply platforms/Windows/bundle/installer.wxs:30]: Sounds reasonable. I'll wire that up in another commit today.

Comment (barnson) [reply platforms/Windows/bundle/installer.wxs:30]: (BTW, build-windows-toolchain.bat doesn't build those packages today, so it's actually going to save the build to move them into existing packages.)

File: platforms/Windows/bundle/installer.wxs
Comment (barnson) [reply platforms/Windows/bundle/installer.wxs:7]: You're right; either architecture would upgrade its own and the other on an Arm64 machine. If making them side-by-side is a real scenario, then they need their own upgrade codes and some directory reorganization so they don't overlap. Windows Installer supports overlap for non-PE files like SDK headers but all the host binaries would need architecture-specific directories to avoid overlap. We can figure out the directories as part of the per-user work if you want it -- especially as we're already messing around with how the tree is organized by moving it out of Program Files.

Comment (tristanlabelle) [reply platforms/Windows/bundle/installer.wxs:7]: @compnerd Do you think we care about this?

**Review by tristanlabelle — APPROVED** [review]

File: platforms/Windows/bld/bld.wxs
Comment (compnerd) [inline platforms/Windows/bld/bld.wxs:331]: Nice I was curious about this.  We likely want to have the same pattern everywhere.

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
Kicked off a new CI run, 🤞 it passes!

**Comment by compnerd** [comment]
The build failed: https://ci-external.swift.org/job/swift-PR-build-toolchain-windows/704/consoleText

Seems that the path to the clang headers is getting truncated @barnson?

**Comment by barnson** [comment]
D'oh. Fixed.

**Comment by compnerd** [comment]
New run kicked off: https://ci-external.swift.org/job/swift-PR-build-toolchain-windows/705/

**Comment by barnson** [comment]
Embedded cabinets match what build-windows-toolchain.bat expects. We should revisit if we can prefer the bundle over the MSIs.

**Comment by compnerd** [comment]
Re-running https://ci-external.swift.org/job/swift-PR-build-toolchain-windows/706/

**Comment by barnson** [comment]
💯

**Comment by compnerd** [comment]
@barnson why is installer.exe 640MB?  That seems like a pretty hefty size increase from 399MB.

**Comment by barnson** [comment]
Silent install doesn't require any gesture to agree with the EULA. Is it needed? I can default the checkbox to checked or get rid of it entirely when I do the custom theme.

**Comment by compnerd** [comment]
> Silent install doesn't require any gesture to agree with the EULA. Is the checkbox needed? I can default the checkbox to checked or get rid of it entirely when I do the custom theme.

No, I don't think that the checkbox is needed.  We can deal with that aspect later :).

**Comment by compnerd** [comment]
https://ci-external.swift.org/job/swift-PR-build-toolchain-windows/708/ build succeeded, seems reasonable (should smoke test it).  Need to wait for the parallel change to be ready before we can merge this.

**Pull request merged**
